pkg/lint/rules/description_must_not_contain_illegal_characters.go

```diff
@@ -4,53 +4,40 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/santhosh-tekuri/jsonschema/v5"
-
 	"github.com/giantswarm/schemalint/pkg/lint"
+	"github.com/giantswarm/schemalint/pkg/lint/utils"
 	"github.com/giantswarm/schemalint/pkg/schemautils"
 )
 
 type DescriptionMustNotContainIllegalCharacters struct{}
 
 var descriptionIllegalCharacters = []string{"\n", "\r", "\t", "  "}
 
-func (r DescriptionMustNotContainIllegalCharacters) Verify(schema *jsonschema.Schema) []string {
-	return lint.RecursePropertiesWithDescription(schema, checkDescriptionDoesNotContainIllegalCharacters)
-}
-
-func checkDescriptionDoesNotContainIllegalCharacters(schema *jsonschema.Schema) []string {
-	ruleViolations := []string{}
+func (r DescriptionMustNotContainIllegalCharacters) Verify(schema *schemautils.ExtendedSchema) lint.RuleResults {
+	ruleResults := &lint.RuleResults{}
 
-	isInvalid, containedIllegalChars := containsIllegalCharacter(schema.Description)
-	if isInvalid {
-		ruleViolations = append(ruleViolations, fmt.Sprintf(
-			"Property '%s' description must not contain %s",
-			schemautils.GetConciseLocation(schema),
-			strings.Join(containedIllegalChars, "', '"),
-		))
-	}
+	propertyAnnotationsMap := utils.BuildPropertyAnnotationsMap(schema).WhereDescriptionsExist()
 
-	if containsLeadingOrTrailingSpace(schema.Description) {
-		ruleViolations = append(ruleViolations, fmt.Sprintf(
-			"Property '%s' description must not contain leading or trailing spaces",
-			schemautils.GetConciseLocation(schema),
-		))
+	for path, annotations := range propertyAnnotationsMap {
+		description := annotations.GetDescription()
+		if isInvalid, containedIllegalChars := containsIllegalCharacter(description); isInvalid {
+			ruleResults.Add(fmt.Sprintf("Property '%s' description must not contain illegal characters: %s", path, containedIllegalChars))
+		}
+		if containsLeadingOrTrailingSpace(description) {
+			ruleResults.Add(fmt.Sprintf("Property '%s' description must not contain leading or trailing spaces", path))
+		}
 	}
-
-	return ruleViolations
+	return *ruleResults
 }
 
 func containsIllegalCharacter(s string) (contains bool, containedIllegalCharacters []string) {
-	contains = false
-
 	for _, illegalCharacter := range descriptionIllegalCharacters {
 		if strings.Contains(s, illegalCharacter) {
-			contains = true
 			containedIllegalCharacters = append(containedIllegalCharacters, illegalCharacter)
 		}
 	}
 
-	return contains, containedIllegalCharacters
+	return len(containedIllegalCharacters) > 0, containedIllegalCharacters
```

Comment:
tiny thing but if named return values are used (e.g `contains` in this function), we should return the actual named variable IMO.

Reply:
You are right, thanks for pointing that out. I've modified the function to make it easier to read.